src/query/src/dist_plan/analyzer.rs

```diff
@@ -223,7 +239,7 @@ impl PlanRewriter {
             Commutativity::Commutative => {}
             Commutativity::PartialCommutative => {
                 if let Some(plan) = partial_commutative_transformer(plan) {
-                    self.update_column_requirements(&plan);
+                    self.update_column_requirements(&plan, self.level - 1);
```

Comment:
Why do we use `level - 1` instead of `level`?

Reply:
> Why do we use `level - 1` instead of `level`?

Because plan is parent of node that might expand